### DISCUSSION TITLE
How to get JSON object in RunjavaScript task

### DISCUSSION THREAD

**Discussion by neerajmathur** [elsa-workflows/elsa-core#4435] [Q&A]
**Input** 

`{
  "SomeDocument": {
    "Title": "About Elsa Workflows",    
  }
}`

**Workflow**
```
builder
	.HttpEndpoint(activity => activity
			.WithPath("/hello-world")
			.WithMethod(HttpMethod.Post.Method)
			.WithReadContent())
			.WithName("MyHttpEndpoint")
	.RunJavaScript(" const inputStepPayload = `${activities.MyHttpEndpoint.Output().Body.SomeDocument.Title}`;  return inputStepPayload; ")
	.WriteHttpResponse(_ => HttpStatusCode.OK, context => $"{context.Input!}", _ => "text/html");
```

**Result**
`About Elsa Workflows`

------------------------------------------------------------------------------------------------------------
**However if I use following**

```
builder
	.HttpEndpoint(activity => activity
			.WithPath("/hello-world")
			.WithMethod(HttpMethod.Post.Method)
			.WithReadContent())
			.WithName("MyHttpEndpoint")
	.RunJavaScript(" const inputStepPayload = `${activities.MyHttpEndpoint.Output().Body.SomeDocument}`;  return inputStepPayload; ")
	.WriteHttpResponse(_ => HttpStatusCode.OK, context => $"{context.Input!}", _ => "text/html");
```

**This results in** 

`System.Dynamic.ExpandoObject`

![image](https://github.com/elsa-workflows/elsa-core/assets/6531916/860d89d7-4b74-49a3-8be1-c37ff4d9eb17)

**How to access an Object from a JSON payload inside RunJavaScript?**

**Comment by sfmskywalker** [comment]
Hi @neerajmathur ,

There's a small issue with your second JavaScript:

```javascript
const inputStepPayload = `${activities.MyHttpEndpoint.Output().Body.SomeDocument}`;  return inputStepPayload;
```

Notice that you're creating a string value using the backticks. This will cause the SomeDocument object (which is an ExpandoObject) to be written as its .NET type name.

What you want to do instead, is return the actual SomeDocument object, not stringified, like this:

```javascript
const inputStepPayload = activities.MyHttpEndpoint.Output().Body.SomeDocument; return inputStepPayload;
```

Or simply:

```javascript
activities.MyHttpEndpoint.Output().Body.SomeDocument
```

That will give you the expected response:

```text
{"Title":"About Elsa Workflows"}
```
